scripts/firedrake-install

```diff
@@ -140,13 +174,23 @@ else:
     args.disable_ssh = False
     args.show_petsc_configure_options = False
     args.adjoint = False
-    args.adjoint = args.adjoint or args.install_adjoint
     args.slope = False
     args.slope = args.slope or args.install_slope
     args.slepc = False
     args.slepc = args.slepc or args.install_slepc
     args.virtualenv_name = False
     args.virtualenv_name = args.virtualenv_name or "firedrake"
+    args.packages = args.packages or []
+
+    config = None
+    if not args.ignore_saved_config:
+        try:
+            import firedrake_configuration
+            config = firedrake_configuration.get_config()
+        except:
+            config = FiredrakeConfiguration(args)
+    if not config:
+        config = FiredrakeConfiguration(args)
```

Comment:
Command-line options should override saved configuration.  So after the config is built, one should parse the options into it. So we should do:

```python
class LegacyStage(object):
    pass
args = LegacyState()
args.developer = False
...

try:
   import firedrake_configuration
   config = firedrake_configuration.get_config()
except AppropriateError:
   # Get default config from legacy state
   config = FiredrakeConfiguration(args)

# Now we need to override stored config with commandline args
config.update(parser.parse_args())
```